src/controllers/v1/attachment/__tests__/index.test.ts

```diff
@@ -0,0 +1,304 @@
+import { expect } from 'chai'
+import sinon from 'sinon'
+import mockdate from 'mockdate'
+
+import { attachment } from '../index.js'
+import { Env } from '../../../../env.js'
+import Database from '../../../../lib/db/index.js'
+import { octetExample, jsonExample } from './fixtures.js'
+import Ipfs from '../../../../lib/ipfs.js'
+import { BadRequest, InternalServerError, NotFound } from '../../../../lib/error-handler/index.js'
+import { AttachmentRow } from '../../../../lib/db/types.js'
+
+mockdate.set('2020-10-10')
+
+describe('v1/attachment', () => {
+  let response: any
+  let controller: attachment
+  const database: Database = new Database()
+  const ipfs: Ipfs = new Ipfs(new Env())
+
+  const stubs = {
+    getFile: sinon
+      .stub(ipfs, 'getFile' as any)
+      .callsFake(async () => ({ blob: { arrayBuffer: () => '0'.repeat(1024) }, filename: 'a' })),
+    addFile: sinon.stub(ipfs, 'addFile' as any).callsFake(async () => 'QmXVStDC6kTpVHY1shgBQmyA4SuSrYnNRnHSak5iB6Eehn'),
+    get: sinon.stub(database, 'get').callsFake(async () => [octetExample]),
+    update: sinon.stub(database, 'update' as any).callsFake(async (_, updates: any) => [
+      {
+        ...octetExample,
+        ...updates,
+      },
+    ]),
+    insert: sinon
+      .stub(database, 'insert' as any)
+      .callsFake((_, data: any) => [{ ...data, created_at: new Date(), id: 'attachment-insert-test' }]),
+  }
+
+  before(() => {
+    controller = new attachment(database, ipfs)
+  })
+
+  after(() => {
```

Comment:
should this be an `afterEach`?

Reply:
Mmm... this is just a safety to no leave hanging stubs that could cause some fleakiness in the future

Reply:
well yes but that flakyness could happen if oone test fails and doesn't clean up right? This is just ensuring we leave a clean environment between tests surely?